src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,80 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
 public class WorkWithFile {
+    static final String COMMA_SEPARATOR = ",";
+    static final int OPERATION_TYPE = 0;
+    static final int AMOUNT = 1;
+    static final String SUPPLY = "supply";
+    static final String BUY = "buy";
+    static final String RESULT = "result";
+    static final int ZERO = 0;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        List<String> stringList = readFile(fromFileName);
+        StringBuilder report = createReport(stringList, toFileName);
+        writeToFile(report, toFileName);
+    }
+
+    private List<String> readFile(String fromFileName) throws RuntimeException {
+        List<String> fromFileToList = new ArrayList<>();
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String readLine = reader.readLine();
+            while (readLine != null) {
+                fromFileToList.add(readLine);
+                readLine = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return fromFileToList;
+    }
+
+    private StringBuilder createReport(List<String> fromFileToList, String toFileName) {
```

Comment:
```suggestion
    private String createReport(List<String> fromFileToList) {
```
you should not put file name here

just remove true flag in writeToFile method